libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java

```diff
@@ -412,7 +454,7 @@ public static String decryptBytes(byte[] data, byte[] key, byte[] iv) {
 
     private static byte[] generateInitVector() throws NoSuchAlgorithmException {
```

Comment:
It appears this is only ever called once, when we first generate the encryption key.  Is that right?  Since GCM (unlike AES) has the two-time-pad/many-time-pad problem shouldn't we be generating a new IV every time?

Reply:
@brandonpage Actually, it looks like a new IV is generated every time we want to encrypt.

It's called from any call to `encrypt(data, key)`:
```
    /**
     * Encrypts data with key using AES/GCM/NoPadding.
     *
     * @param data Data.
     * @param key Base64 encoded 256 bit key or null (to leave data unchanged).
     * @return Encrypted data.
     */
    public static String encrypt(String data, String key) {
        try {
            return encrypt(data, key, generateInitVector());
        } catch (Exception ex) {
            SalesforceAnalyticsLogger.w(null, TAG, "Error during encryption", ex);
        }
        return null;
    }
```
And each call to it generates a new IV:
```
    private static byte[] generateInitVector() throws NoSuchAlgorithmException {
        final SecureRandom random = SecureRandom.getInstance(SHA1PRNG);
        byte[] iv = new byte[12];
        random.nextBytes(iv);
        return iv;
    }
```
It also gets called from `getEncryptingCipher(key)`, but that also generates a new IV every time:
```
    /**
     * Returns initialized cipher for encryption with an IV automatically generated.
     *
     * @param encryptionKey Encryption key.
     * @return Initialized cipher.
     */
    public static Cipher getEncryptingCipher(String encryptionKey)
            throws NoSuchAlgorithmException, InvalidAlgorithmParameterException, InvalidKeyException {
        final byte[] keyBytes = Base64.decode(encryptionKey, Base64.DEFAULT);
        return getEncryptingCipher(keyBytes, generateInitVector());
    }
```

Reply:
Ah, I missed that `generateInitVector()` was called in called at the top level `encrypt` (since it wasn't changed in this PR).  👍